### DISCUSSION TITLE
[分享] 使用Railway部署memos

### DISCUSSION THREAD

**Discussion by redisread** [usememos/memos#1239] [Show and tell]
Railway 地址：[https://railway.app/](https://railway.app/)
memos 地址：[https://github.com/usememos/memos](https://github.com/usememos/memos)

## 注册并且登录 Railway

注册就按照注册流程进行注册即可，进入首页：
![image.png](https://cos.jiahongw.com/PicGo/20230303222909.png)

每个月 Railway 是有 5 美元的免费额度的，在右上角下拉的 `Project Usage` 可以查看

![image.png](https://cos.jiahongw.com/PicGo/20230303223032.png)

## 新建项目

选择新建项目按钮，然后选择 `Deploy from Github repo` 。然后会让你登录 Github 账号并且验证。
![image.png](https://cos.jiahongw.com/PicGo/20230303223218.png)

验证完成之后选择需要部署的代码仓库（注意，这里是需要自己的代码仓库，需要将 memos 的 Github 仓库 fork 一份到自己的仓库下）。选择仓库下的 `memos` 仓库进行立即部署：
![image.png](https://cos.jiahongw.com/PicGo/20230303223541.png)

立即部署之后还不能直接使用，还需要进行后面的一些配置。

## 配置端口

点击 `memos` 项目，然后配置 `Variables` 中添加变量PORT，值为5230
![image.png](https://cos.jiahongw.com/PicGo/20230303223937.png)

在进行完上面的端口配置之后，Railway会自动进行重新部署，部署完成之后就可以访问了。访问的域名在 `Deployment` 的最新部署记录中：
![image.png](https://cos.jiahongw.com/PicGo/20230303224156.png)

![image.png](https://cos.jiahongw.com/PicGo/20230303224231.png)

之后只要有新代码的提交或者是执行 `fork sync`，都能够触发Railway的自动部署。

参考文档：[使用Railway部署memos – ✏VictorHong's Blog](https://hugo.jiahongw.com/zh/20230303/railway-deploy-memos/)

**Comment by Theergold** [comment]
然后一个更新数据直接没了(

**Comment by Katou-Kouseki** [comment]
更新就寄了呀

**Comment by redisread** [comment]
[2023-07-09]更新，使用 Railway 的 Volume 可不丢失数据进行更新，下面的具体的指导教程

1. 参考[https://docs.railway.app/reference/volumes](https://docs.railway.app/reference/volumes)这里的说明，使用`volume`功能需要优先登机会员，参考[Priority Boarding | Railway Docs](https://docs.railway.app/reference/priority-boarding)可知道加入的方法即连接官方的`Discord`即可。
   - 访问您的 Railway General Settings
   - 滚动到 Account Settings，连接您的 Railway 帐户到 Discord
   - 在 Discord 中打开任何频道，输入“/beta”命令并按照提示操作
   - 现在您应该可以访问“#priority-boarding”频道，并且您的帐户设置中应该显示新的 Priority Boarding 状态
   - 从此时起，您将自动启用 Priority Boarding 功能
     ![8xT7xL](https://cos.jiahongw.com/uPic/8xT7xL.jpg)
2. 在项目界面，按`Commond + K`新建一个`volume`
   ![GYpOOq](https://cos.jiahongw.com/uPic/GYpOOq.png)
   volume 的地址填写为`/var/opt/memos`，名字随便。
3. 因为 VOLUME 关键字在 Railway 被禁用了，需要修改 memos 的 Dockerfile 文件，将 VOLUME 那一行删除
   ![ojHt1B](https://cos.jiahongw.com/uPic/ojHt1B.png)
4. 重新部署，记得配置 `Variables` 中添加变量 PORT，值为 5230，就可以了。

**Reply by Theergold** [reply]
在注册管理员账户的时候遇到 `Incorrect login credentials, please try again` 不知道是哪一步出问题了

**Comment by blhorizon** [comment]
自已memos的数据可以提取到本地吗

**Comment by woshichenghaibo** [comment]
> 自已memos的数据可以提取到本地吗
可以备份吧？
